### PR TITLE
notifications overhaul

### PR DESCRIPTION
**Pull request by april83c**
- notifications can have their own text instead of the message being hardcoded in the template json
- [broadcaster](https://github.com/osuplace/broadcaster/) compatibility (use broadcaster API to get topics, etc)
- this does NOT have the logic for not connecting to a websocket if we aren't subscribed to any of their topics

### CODE REVIEW

**Comment by april83c** [comment]
i'm marking this as ready just in case i don't get broadcaster finished before endu has to go, BUT it is untested. i'll try to test it and make sure it works before endu has to go though

File: dist/templateManager.user.js [hunk 1]
```diff
@@ -576,11 +576,70 @@
...
+                });
+                ws.addEventListener('message', async (event) => {
+                    // https://github.com/osuplace/broadcaster/blob/main/API.md
+                    let data = JSON.parse(await event.data.text());
+                    if (data.e == 1) {
+                        if (!data.t || !data.c) {
+                            console.error(`Malformed event from ${serverUrl}: ${data}`);
+                        }
+                        if (this.enabledNotifications.includes(`${serverUrl}??${data.t}`)) {
+                            this.notificationManager.newNotification(serverUrl, data.c);
```
Comment (LittleEndu) [inline dist/templateManager.user.js:623]: change `serverUrl` to `new URL(serverUrl).hostname` so it looks a bit cleaner to the user

Comment (LittleEndu) [reply dist/templateManager.user.js:622]: why did I make this comment in the userscript not typescript 💀

**Comment by april83c** [comment]
ready for review

**Review by MiraiSubject — APPROVED** [review]
LGTM

**Pull request merged**